.github/workflows/main.yml

```diff
@@ -38,13 +38,27 @@ jobs:
           # A directory to store and save the cache
           path: ~/.opam
           # An explicit key for restoring and saving the cache
-          key: ${{ matrix.os }}-${{ matrix.ocaml-version }}-${{ hashFiles('*.opam') }}-build
-      - name: Set up OCaml ${{ matrix.ocaml-version }}
-        uses: avsm/setup-ocaml@v1
+          key: ${{ matrix.os }}-${{ matrix.ocaml-compiler }}-${{ hashFiles('*.opam') }}-build
```

Comment:
The explicit use of `actions/cache@v2` is no longer needed from v2, so it's best to remove it from the workflow.

Reply:
Noted, thank you !